dlt/extract/hints.py

```diff
@@ -68,11 +69,17 @@ class TResourceHints(TResourceHintsBase, total=False):
 
 
 class HintsMeta:
-    __slots__ = ("hints", "create_table_variant")
+    __slots__ = ("hints", "create_table_variant", "data_item_format")
```

Comment:
I opted to add a slot on the default hints, I could also create a subclass similar to the file import, but for me this solution somehow makes more sense, since it is kind of a hint, but does not need to go into the schema.

Reply:
I think you do not need to modify `HintsMeta`. Just create a container class in which you'll yield models ie
```py
class ModelStr(str):
   pass
```
and yield it. Are you 100% sure we do not need any model properties to be stored in the file? or we plan to pass those via TableSchema? ie. look here:
https://docs.getdbt.com/reference/model-configs (materialization)
https://docs.getdbt.com/docs/build/incremental-strategy (this is our write disposition + primary key) heh cool

Reply:
I have done it this way now.